### PR TITLE
Add UnsignedAddWithOverFlow in IlBuilder

### PR DESCRIPTION
**Pull request by ShuyuLi-ca**
Signed-off-by: Shuyu Li <shuyuli@ca.ibm.com>

### CODE REVIEW

File: compiler/ilgen/IlBuilder.cpp [hunk 1]
```diff
@@ -1212,6 +1217,30 @@ IlBuilder::AddWithOverflow(TR::IlBuilder **handler, TR::IlValue *left, TR::IlVal
...
+   {
+   appendBlock(); 
+
+   TR::Node *leftNode = loadValue(left);
+   TR::Node *rightNode = loadValue(right);
+   TR::ILOpCodes op;
+   if (left->getSymbol()->getDataType() == TR::Address)
+      {    
+      if (right->getSymbol()->getDataType() == TR::Int32)
+         op = TR::aiadd;
```
Comment (mstoodle) [inline compiler/ilgen/IlBuilder.cpp]: shouldn't this be iuadd ?
and the following on luadd?

File: compiler/ilgen/IlBuilder.cpp [hunk 2]
```diff
@@ -1212,6 +1217,30 @@ IlBuilder::AddWithOverflow(TR::IlBuilder **handler, TR::IlValue *left, TR::IlVal
    }
 
 TR::IlValue *
+IlBuilder::UnsignedAddWithOverflow(TR::IlBuilder **handler, TR::IlValue *left, TR::IlValue *right)
```
Comment (mstoodle) [inline compiler/ilgen/IlBuilder.cpp]: please add a ILB_REPLAY line as well for this call.

**Comment by mstoodle** [comment]
This PR should have an issue and a more descriptive commit message.

File: compiler/ilgen/IlBuilder.cpp (hunk 2 above)
Comment (ShuyuLi-ca) [inline compiler/ilgen/IlBuilder.cpp]: Thanks for all the suggestions. @mstoodle Could you please take a look? Any suggestions will be appreciate. :)

File: compiler/ilgen/IlBuilder.cpp [hunk 3]
```diff
@@ -1212,6 +1217,31 @@ IlBuilder::AddWithOverflow(TR::IlBuilder **handler, TR::IlValue *left, TR::IlVal
...
+      else if (right->getSymbol()->getDataType() == TR::Int64)
+         op = TR::aluadd;
+      else 
+         TR_ASSERT(0, "the right child type must be either TR::Int32 or TR::Int64 when the left child of Add is TR::Address\n");
+      }    
+   else op = unsignedAddOpCode(leftNode->getDataType());
+
+   TR::IlValue *addValue = operationWithOverflow(op, leftNode, rightNode, handler);
+   TraceIL("IlBuilder[ %p ]::%d is UnsignedAddWithOverflow %d + %d\n", this, addValue->getCPIndex(), left->getCPIndex(), right->getCPIndex());
+   ILB_REPLAY("%s = %s->UnsignedAddWithOverflow(%s, %s);", REPLAY_VALUE(addValue), REPLAY_BUILDER(this), REPLAY_BUILDER(*handler), REPLAY_VALUE(left), REPLAY_VALUE(right));
```
Comment (mstoodle) [inline compiler/ilgen/IlBuilder.cpp]: The function should have 3 operands (you only have "(%s, %s);" )

Comment (ShuyuLi-ca) [inline compiler/ilgen/IlBuilder.cpp]: Sorry for that. Just fixed them already, :)

**Pull request merged**